tox.ini

```diff
@@ -1,49 +1,56 @@
 [tox]
 minversion = 3.5.0
-skipdist = True
+skipsdist = True
 skip_missing_interpreters = True
 isolated_build = True
+toxworkdir = {env:TOXWORKDIR:/tmp/{env:USER}/tox}
 envlist = lint,server,agent
 
 [testenv]
 usedevelop = True
-install_command = pip install -U {opts} {packages}
+install_command = pip install --no-cache-dir --progress-bar off --prefix={envdir} {opts} {packages}
+passenv =
+    PY_COLORS
+    NO_COLORS
+    TERM
+    USER
 setenv =
-   VIRTUAL_ENV={envdir}
+    VIRTUAL_ENV = {envdir}
+    XDG_CACHE_HOME = {envdir}
+    SKIP_GENERATE_AUTHORS = 1
+    SKIP_WRITE_GIT_CHANGELOG = 1
 envars =
-   PATH = {env:PATH}:{toxinidir}/bin
+    PATH = {env:PATH}:{toxinidir}/bin
+commands_pre =
+    {toxinidir}/utils/detox {envdir}
```

Comment:
Aside from the most excellent name, I don't think the `detox` script is worth keeping:  it wraps a single command, and the effort of locating the script and feeding it the appropriate input is greater than the encapsulation is worth.  I would just replace this line with the command invocation and be done with it:
```
    find {envdir} -type f -path '*/__pycache__/*' -name '*.py[c|o]' -delete
```

Reply:
I originally moved to the `detox` script so that I could run the find command myself separate from `tox` to work with and debug problems.  It was always a pain to open the `tox` file and grab the command instead of just invoking `utils/detox`.  And further, I was hoping that we could place other cleanups in there as well.